core-blocks/archives/block.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import {
+	PanelBody,
+	ServerSideRender,
+	ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	InspectorControls,
+	BlockAlignmentToolbar,
+	BlockControls,
+} from '@wordpress/editor';
+
+import './editor.scss';
+
+class ArchivesBlock extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.toggleShowPostCounts = this.toggleShowPostCounts.bind( this );
+		this.toggleDisplayAsDropdown = this.toggleDisplayAsDropdown.bind( this );
+	}
+
+	toggleShowPostCounts() {
+		const { attributes, setAttributes } = this.props;
+		const { showPostCounts } = attributes;
+
+		setAttributes( { showPostCounts: ! showPostCounts } );
+	}
+
+	toggleDisplayAsDropdown() {
+		const { attributes, setAttributes } = this.props;
+		const { displayAsDropdown } = attributes;
+
+		setAttributes( { displayAsDropdown: ! displayAsDropdown } );
+	}
+
+	render() {
+		const { attributes, isSelected, setAttributes } = this.props;
+		const { align, showPostCounts, displayAsDropdown } = attributes;
+
+		const inspectorControls = isSelected && (
+			<InspectorControls key="inspector">
+				<PanelBody title={ __( 'Archives Settings' ) }>
+					<ToggleControl
+						label={ __( 'Show post counts' ) }
```

Comment:
We probably should make this title case for consistency with other inspector control labels in Gutenberg.

```js
label={ __( 'Show Post Counts' ) }
```